src/Lachain.Consensus/ThresholdKeygen/Data/State.cs

```diff
@@ -66,15 +69,17 @@ public static State FromBytes(ReadOnlyMemory<byte> bytes)
                 .Batch(Fr.ByteSize)
                 .Select(x => Fr.FromBytes(x.ToArray()))
                 .ToArray();
-            var acks = bytes.Slice(8 + cLen + n * Fr.ByteSize).ToArray()
+            var acks = bytes.Slice(8 + cLen + n * Fr.ByteSize, n).ToArray()
                 .Select(b => b != 0)
                 .ToArray();
+            var confirmation = bytes.Slice(8 + cLen + n * Fr.ByteSize + n).ToArray();
```

Comment:
Have you tested new node synchronization from the genesis block after hardfork? I mean setup like this: start a network with old nodes,  than update them at some point to the new nodes,  generate some anount of blocks,  then delete blockchain data for one of the new blocks and try to sync it with other nodes.

Reply:
Tested.